src/lib/tls/tls_session.h

```diff
@@ -139,8 +153,11 @@ class BOTAN_PUBLIC_API(2,0) Session final
 
       /**
       * Get the session identifier
+      *
+      * Note that for TLS 1.3 sessions this will be equal to the session ticket
+      * and might therefore be a up to 2^16-1 bytes long.
       */
-      const std::vector<uint8_t>& session_id() const { return m_identifier; }
+      const std::vector<uint8_t>& session_id() const;
```

Comment:
This is worth noting! Probably even more so for users of the library, as they might need to deal with the fact that session IDs are suddenly not 32-byte buffers anymore but potentially much larger. Will there be a "migration guide to Botan 3.0" of some sort where this could be mentioned?

TLS 1.3 removes the concept of session IDs entirely, because resumption was merged with Pre-Shared-Key negotiation. Instead PSK negotiations that stem from session tickets are identified by the opaque session ticket buffer itself. However, our `Session_Manager` API relies on the session id as a database index to retrieve stored sessions on the server side.

I considered "inventing" session IDs by simply hashing the TLS 1.3 session tickets, but found that to be intransparent magic that might better be implemented by user's application code. Especially when we want to enable library users to actually use the session ticket as a means to keep encrypted state on the client side.

Thoughts?

Reply:
For now I just added a remark in the migration guide. Thanks for pointing out this document, by the way!